### DISCUSSION TITLE
Hierarchical data / Tree

### DISCUSSION THREAD

**Discussion by Fee0** [glideapps/glide-data-grid#256] [Q&A]
Hi,

Can the grid be used to represent hierarchical / tree like data. E.g. a file system structure?
I didn't found anything related.
If its not supported, how difficult would it be to add it myself?
I imagine it like this: An extra column in my data for the hierarchie and a handler for oncellclick which hides oder rows that are under the current row in the hierarchie. 

any ideas`?

**Comment by jassmith** [comment]
I've been meaning to work on making this viable. Currently the biggest issues is the disclosure indicator needed. It's easy to add. If you want to collaborate on doing this together I'm more than happy to start putting together the basic framework. I estimate this will take between 1 and 2 days to handle the simple case, and 1 to 2 weeks to get more general support.

**Reply by Fee0** [reply]
thank you for the estimate that helps me a lot. For now I'm doing an evaluation about the capabilities of different grids to see what fits my requirements the best. Since the grid is canvas based it not simply about putting another dom element into a cell I guess? Its more like defining pixel values for the indicator at a certain coordinate in the grid? Is there anything that describes the technique you used to implement this grid? I think I need to read up a bit how to deal with canvas ui first.
